improver/utilities/enforce_consistency.py

```diff
@@ -0,0 +1,130 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Provides utilities for enforcing consistency between forecasts."""
+import warnings
+
+import numpy as np
+from iris.cube import Cube
+
+from improver import PostProcessingPlugin
+
+
+class EnforceConsistentForecasts(PostProcessingPlugin):
+    """Enforce that the forecasts provided are no less than or no greater than some
+    linear function of a reference forecast. For example, wind speed forecasts may be
+    provided as the reference forecast for wind gust forecasts with a requirement that
+    wind gusts are not less than 110% of the corresponding wind speed forecast. Wind
+    speed forecasts of 10 m/s will mean that the resulting wind gust forecast will be
+    at least 11 m/s.
+    """
+
+    def __init__(
+        self,
+        additive_amount: float = 0.0,
+        multiplicative_amount: float = 1.0,
+        comparison_operator: str = ">=",
+        diff_for_warning: float = None,
+    ) -> None:
+        """
+        Initialise class for enforcing consistency between a forecast and a linear
+        function of a reference forecast.
+
+        Args:
+            additive_amount: The amount to be added to the reference forecast prior to
+                enforcing consistency between the forecast and reference forecast. If
+                both an additive_amount and multiplicative_amount are specified then
+                addition occurs after multiplication.
+            multiplicative_amount: The amount to multiply the reference forecast by
+                prior to enforcing consistency between the forecast and reference
+                forecast. If both an additive_amount and multiplicative_amount are
+                specified then addition occurs after multiplication.
+            comparison_operator: Determines whether the forecast is enforced to be not
+                less than or not greater than the reference forecast. Valid choices are
+                ">=", for not less than, and "<=" for not greater than.
+            diff_for_warning: If assigned, the plugin will raise a warning if any
+                absolute change in forecast value is greater than this value.
+        """
+
+        self.additive_amount = additive_amount
+        self.multiplicative_amount = multiplicative_amount
+        self.comparison_operator = comparison_operator
+        self.diff_for_warning = diff_for_warning
+
+    def process(self, forecast: Cube, reference_forecast: Cube) -> Cube:
+        """
+        Function to enforce that the values in forecast_cube are not less than or not
+        greater than a linear function of the corresponding values in
+        reference_forecast.
+
+        Args:
+            forecast: A forecast cube of probabilities or percentiles
```

Comment:
I did consider this, and I think the output as it currently exists will work with realisation forecasts.

The reason I haven't made this functionality explicit is that I'm not sure enforcing consistency between realisation forecasts necessarily makes sense. There is not necessarily an ordering to the realisations in an ensemble, so that we don't necessarily want the 5th ensemble member of the forecast to be consistent with the corresponding ensemble member of a reference forecast. 

I haven't explicitly disallowed this functionality currently as a hypothetical user might want to enforce consistency between 2 ordered ensembles, which I don't think would be unreasonable.